### ISSUE TITLE
Unable to install after Xcode update

### ISSUE THREAD

**Issue by tarasmatsyk** [python-pillow/Pillow#6725]
Thank you for an awesome project, guys! (no sarcasm here)

### What did you do?
I am trying to install Pillow after long awaited xcode update (always fun and exciting)
`pip install Pillow~=8.2.0`

### What did you expect to happen?
Pillow installed

### What actually happened?
Pillow threw at me 175k lines of stacktrace :) 

2 errors:
```
   In file included from src/_imagingft.c:26:
      In file included from /opt/homebrew/Cellar/freetype/2.12.1/include/freetype2/freetype/freetype.h:24:
      In file included from /opt/homebrew/Cellar/freetype/2.12.1/include/freetype2/freetype/config/ftconfig.h:48:
      In file included from /opt/homebrew/Cellar/freetype/2.12.1/include/freetype2/freetype/config/mac-support.h:33:
      /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/AvailabilityMacros.h:248:5: error: 'TARGET_OS_OSX' is not defined, evaluates to 0 [-Werror,-Wundef-prefix=TARGET_OS_]
      #if TARGET_OS_OSX
          ^
      /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/AvailabilityMacros.h:250:7: error: 'TARGET_OS_MACCATALYST' is not defined, evaluates to 0 [-Werror,-Wundef-prefix=TARGET_OS_]
      #elif TARGET_OS_MACCATALYST
```

most common warnings:
```
char    *l64a(long);
              ^
                _Nullable 32
      : warning: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Wnullability-completeness]/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdlib.h:207:29: note: insert '_Nullable' if the array parameter may be null
      /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdlib.h
      :206:6:void     lcong48(unsigned short[7]);
                                     ^
                                      _Nullable
       note: insert '_Nonnull' if the pointer should never be null
```
```
unsigned char   *_base;
                              ^
                                _Nonnull
      /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/_stdio.h:138:32: warning: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Wnullability-completeness]
```
```
defined(SWIFT_CLASS_EXTRA) && (!defined(SWIFT_SDK_OVERLAY_PTHREAD_EPOCH) || (SWIFT_SDK_OVERLAY_PTHREAD_EPOCH < 1))
              ^
      /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/pthread.h:563:6: warning: macro expansion producing 'defined' has undefined behavior [-Wexpansion-to-defined]
```

### What are your OS, Python and Pillow versions?

* OS:  macOS, 13.0
* Python:  3.7.13
* Pillow:  Pillow~=8.2.0

Reproducable for the latest Pillow version as well. Would be happy to help with troubleshooting, not sure if it's Apple messed up or I need to configure my system somehow

**Comment by Yay295** [comment]
I found this golang issue with a similar error: https://github.com/golang/go/issues/38876#issuecomment-630589519

The solution was to use `CGO_CPPFLAGS=-Wno-error -Wno-nullability-completeness -Wno-expansion-to-defined -Wno-builtin-requires-header`. It might be something similar for Pillow.

**Comment by tarasmatsyk** [comment]
@Yay295 , he-he, that's exactly what I tried a few minutes before posting this. It was my last attempt to configure clang build with no luck :) 

Appreciate a link here, might give a hint to engineers with better understanding of the issue

**Comment by radarhere** [comment]
How did you install Python 3.7?

From what I can see, the default Python in macOS is Python 3.9. Does this same problem happen if you use the default Python?

**Comment by tarasmatsyk** [comment]
@radarhere, appreciate quick reply. I think my system got into complete mess after xcode update. I reinstalled xcode, python (via brew), pyenv and finally was able to get Pillow installed

I am using python 3.10 from homebrew (latest one) and pyenv to install python 3.7

RE Python 3.9, latest Pillow worked fine on default python. However the issue was the same for py 3.7 installed through pyenv. 

Unfortunately I cannot share the full log as it contains too many references to private projects, however I can assure you that those are were the only errors I managed to find (I spent ~40 minutes to find all errors before googling :D) 

At the moment, the issue has been fixed and I am happy to close it for a lack of reproducing steps. If you feel like it will be useful for other people, would be happy to help with troubleshooting
